discovery/gossiper.go

```diff
@@ -795,78 +812,23 @@ func (d *deDupedAnnouncements) Emit() []msgWithSenders {
-func (d *AuthenticatedGossiper) resendAnnounceSignatures() error {
-	peerMsgsToResend, err := d.cfg.MessageStore.Messages()
+// resendPendingMsgs retrieves and sends all of the messages within the
+// gossiper's message store that should be reliably sent to their respective
+// peers. Before sending each message, a check is done to determine if the
+// message is stale to prevent sending it.
+func (d *AuthenticatedGossiper) resendPendingMsgs() error {
+	// Fetch all of the peers for which we have pending messages for and
+	// spawn a peerMsgHandler for each. Once the peer is seen as online, all
+	// of the pending messages will be sent.
+	peers, err := d.cfg.MessageStore.Peers()
 	if err != nil {
 		return err
 	}
 
-	// We now iterate over these messages, resending those that we don't
-	// have the full proof for, deleting the rest.
-	for peer, msgsToResend := range peerMsgsToResend {
-		pubKey, err := btcec.ParsePubKey(peer[:], btcec.S256())
-		if err != nil {
-			return err
-		}
-
-		for _, msg := range msgsToResend {
-			msg := msg.(*lnwire.AnnounceSignatures)
-
-			// Check if the full channel proof exists in our graph.
-			chanInfo, _, _, err := d.cfg.Router.GetChannelByID(
-				msg.ShortChannelID)
-			if err != nil {
-				// If the channel cannot be found, it is most likely a
-				// leftover message for a channel that was closed.  In
-				// this case we delete it from the message store.
-				log.Warnf("unable to fetch channel info for "+
-					"chanID=%v from graph: %v. Will delete local"+
-					"proof from database",
-					msg.ChannelID, err)
-				err = d.cfg.MessageStore.DeleteMessage(msg, peer)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-
-			// 1. If the full proof does not exist in the graph, it means
-			// that we haven't received the remote proof yet (or that we
-			// crashed before able to assemble the full proof). Since the
-			// remote node might think they have delivered their proof to
-			// us, we will resend _our_ proof to trigger a resend on their
-			// part: they will then be able to assemble and send us the
-			// full proof.
-			if chanInfo.AuthProof == nil {
-				err := d.sendAnnSigReliably(msg, pubKey)
-				if err != nil {
-					return err
-				}
-				continue
-			}
-
-			// 2. If the proof does exist in the graph, we have
-			// successfully received the remote proof and assembled the
-			// full proof. In this case we can safely delete the local
-			// proof from the database. In case the remote hasn't been able
-			// to assemble the full proof yet (maybe because of a crash),
-			// we will send them the full proof if we notice that they
-			// retry sending their half proof.
-			if chanInfo.AuthProof != nil {
-				log.Debugf("Deleting message for chanID=%v from "+
-					"messageStore", msg.ChannelID)
-				err := d.cfg.MessageStore.DeleteMessage(msg, peer)
-				if err != nil {
-					return err
-				}
-			}
-		}
+	for peer := range peers {
+		d.reliableSender.spawnPeerHandler(peer)
```

Comment:
the whole `resendPendingMsgs` logic could be moved to the `reliableSender`?

Reply:
Fixed.